web/src/message_edit.js

```diff
@@ -1070,17 +1070,29 @@ export function save_message_row_edit($row) {
 
                 hide_message_edit_spinner($row);
                 if (xhr.readyState !== 0) {
-                    const message = channel.xhr_error_message(
-                        $t({defaultMessage: "Error editing message"}),
-                        xhr,
-                    );
+                    let message;
+                    let html_content = false;
+                    if (xhr.responseJSON?.code === "TOPIC_WILDCARD_MENTION_NOT_ALLOWED") {
+                        message = $t_html({
+                            defaultMessage:
+                                "You do not have permission to use <b>@topic</b> mentions in this topic.",
+                        });
+                        html_content = true;
+                    } else {
+                        message = channel.xhr_error_message(
+                            $t({defaultMessage: "Error editing message"}),
+                            xhr,
+                        );
+                    }
                     const $container = compose_banner.get_compose_banner_container(
                         $row.find("textarea"),
                     );
                     compose_banner.show_error_message(
                         message,
                         compose_banner.CLASSNAMES.generic_compose_error,
                         $container,
+                        undefined,
+                        html_content,
```

Comment:
I think logic would be just `append_compose_banner_to_banner_list` in the `TOPIC_WILDCARD_MENTION_NOT_ALLOWED` code path, and `compose_banner.show_error_message` in the else clause.

Reply:
Done.